src/modules/auth/components/register/Register.tsx

```diff
@@ -81,145 +181,126 @@ export function Register({ search }: RegisterProps) {
-            autoFocus
-          />
-          <EmailInput
-            required
-            value={email}
-            setValue={setEmail}
-            id={EMAIL_SIGN_UP_FIELD_ID}
-            disabled={Boolean(invitation?.email)}
-            shouldValidate={shouldValidate}
-          />
-          <Stack>
-            <EnableAnalyticsForm
-              enableSaveActions={enableSaveActions}
-              onUpdateSaveActions={(enabled) => setEnableSaveActions(enabled)}
-            />
-
-            <AgreementForm useAgreementForm={agreementFormHook} />
-          </Stack>
-          <ErrorDisplay error={registerError} />
-          <LoadingButton
-            variant="contained"
-            id={REGISTER_BUTTON_ID}
-            loading={isLoadingSignUp || isLoadingMobileSignUp}
-            onClick={handleRegister}
-            fullWidth
-            disabled={!userHasAcceptedAllTerms || !email.length || !name.length}
-          >
-            {t(SIGN_UP_BUTTON)}
-          </LoadingButton>
+              }),
+            },
+            minLength: {
+              value: MIN_USERNAME_LENGTH,
+              message: translateCommon('USERNAME_MIN_LENGTH', {
+                min: MIN_USERNAME_LENGTH,
+              }),
+            },
+            validate: (name) =>
+              MemberConstants.USERNAME_FORMAT_REGEX.test(name.trim()) ||
+              translateCommon('USERNAME_SPECIAL_CHARACTERS'),
+          })}
+          placeholder={t(NAME_FIELD_LABEL)}
+          // todo: Should we not allow users to change their name when creating a count from an invitation ?
+          // disabled={Boolean(invitation?.name)}
+        />
+        <StyledTextField
+          id={EMAIL_SIGN_UP_FIELD_ID}
+          slotProps={{
+            input: {
+              startAdornment: EmailAdornment,
+            },
+          }}
+          variant="outlined"
+          // type="email"
+          error={Boolean(emailError)}
+          helperText={emailError}
+          {...register('email', {
+            required: translateCommon('REQUIRED'),
+            validate: (email) =>
+              isEmail(email, {}) || translateCommon('INVALID_EMAIL'),
+          })}
+          placeholder={t('EMAIL_INPUT_PLACEHOLDER_REQUIRED')}
+          // do not allow to modify the email if it was provided
+          disabled={Boolean(initialData.email)}
+        />
+        <Stack>
+          <EnableAnalyticsForm control={control} />
+          <AgreementForm control={control} />
         </Stack>
-      </FormControl>
-      <Typography>{t(SIGN_IN_LINK_TEXT)}</Typography>
-      <Link to="/auth/login" search={search}>
-        {t(SIGN_IN_LINK_TEXT_BUTTON)}
-      </Link>
+        <ErrorDisplay error={registerError} />
+        <LoadingButton
+          type="submit"
+          variant="contained"
+          id={REGISTER_BUTTON_ID}
+          loading={isLoadingSignUp || isLoadingMobileSignUp}
+          fullWidth
+          disabled={Boolean(Object.keys(errors).length > 0)}
```

Comment:
What I did in builder is the following, isSubmitted being necessary because the initial value before submitting is false. In your case it's the same since `errors` is computed only after submitting. 
```suggestion
          disabled={isSubmitted && !isValid}
```

Reply:
This is interesting, but I tired it and since the button is disabled, it does not show the error message. So not quite what I was going for. But I learned something !